### ISSUE TITLE
[Question] Does cypress-select-tests support Typescript?

### ISSUE THREAD

**Issue by git-malu** [bahmutov/cypress-select-tests#133]
I guess cypress-select-tests does not support Typescript so far.
Because when I try to use cypress-select-tests together with Typescript, I get the error below.

`SyntaxError: 'import' and 'export' may appear only with 'sourceType: module' (17:0)`

My Cypress package works well with Typescript if I comment out this plugin.

`// on('file:preprocessor', selectTestsWithGrep(config));`

**Comment by sagar-ganatra** [comment]
+1, facing the same issue

**Comment by csalazar1** [comment]
@bahmutov any chance you can do a quick fix? Seems like a handful of us are having this issue

**Comment by rvowles** [comment]
It wasn't a quick fix. The library uses falafel to parse the source code and rewrite steps with .skip. I have swapped it to use the typescript compiler (which will of course read javascript) and rewrite it in the same fashion, but it also needs to turn typescript on as this library wipes out the normal cypress ability to process typescript.

**Comment by csalazar1** [comment]
@rvowles was this merged to master already?

**Comment by rvowles** [comment]
No, I haven't pushed it up and I don't think @bahmutov will want it given it swaps from falafel to typescript. Its currently under review where I work.

**Comment by csalazar1** [comment]
Got it. Thank you so much for the update on this

**Comment by rvowles** [comment]
Ok @csalazar1 - my "branch" is over here but I'm not issuing a PR because I don't think @bahmutov will be that interested. https://github.com/rvowles/cypress-select-tests

It does a few things - it always enables typescript, it uses a ts parser for typescript and javascript one for javascript. We only use typescript here, so I left the js one for the tests. 

Further it has a preprocessor that reads the cypress.json file, applies your filtering and updates the cypress.json file into a cypress-grep.json file, which then includes the filtering but removes any files in your spec list that wouldn't be run as they have no tests.

One thing you probably won't like is that it actively removes tests that are not run rather than leaving them pending. Pending on our case implies there is some action that needs to be done on them, confusing the reporting. In typescript, this removes the step and replaces it with a placeholder ("true").

**Comment by csalazar1** [comment]
@rvowles I finally got around to this. It seems to be failing with this error though
<img width="448" alt="error" src="https://user-images.githubusercontent.com/32462093/94383226-484da380-00f4-11eb-9930-5b8f0f1ccf60.png">

Any ideas what it could be?

**Comment by rvowles** [comment]
If it is acorn your issue is that it is treating whatever that file is as JavaScript and parsing it using acorn and falafel.

**Comment by csalazar1** [comment]
Must be using those by default. Would I need to specify a TS parser in tsconfig or something like that

**Comment by biltongza** [comment]
@rvowles we're trying to use your plugin, stock standard config (i.e. `on('file:preprocessor', selectTestsWithGrep(config));`) and we keep getting `SyntaxError: 'import' and 'export' may appear only with 'sourceType: module'` no matter what we change. We've tried messing with tsconfig files but nothing seems to work. How do we tell this thing that we are using typescript files? It's like it isn't finding our tsconfig file (which is in the cypress directory)

**Comment by rvowles** [comment]
Maybe ask this question on my repo instead of here?

**Comment by biltongza** [comment]
![image](https://user-images.githubusercontent.com/2845704/97200389-d2ebf600-17b9-11eb-9bfd-ccbb05048b36.png)
you don't have Issues enabled on your repo :(

**Comment by onimougwo** [comment]
Any update ?

**Comment by estefafdez** [comment]
I would love Typescript to be supported by this plug-in. Thanks for all the hard work on this one!

**Comment by NeetuHundal** [comment]
Does anyone was able to resolve this error SyntaxError: 'import' and 'export' may appear only with 'sourceType: module'? If yes, please guide how did you do?

**Comment by NeetuHundal** [comment]
Does anyone was able to resolve this error SyntaxError: 'import' and 'export' may appear only with 'sourceType: module'? If yes, please guide how did you do.Really appreciate. Thank you.
